packages/es-client/search.js

```diff
@@ -91,29 +111,148 @@ const esMetricsConfig = () => {
   };
 };
 
+/**
+ * Generates a configuration for Elasticsearch based on the environment
+ * and provided parameters.
+ *
+ * @param {string} [host] - The host URL for the Elasticsearch instance.
+ * @param {boolean} [metrics=false] - A flag indicating whether metrics are enabled.
+ * @returns {Promise<[Object, import('@aws-sdk/types').Credentials | undefined]>} A
+ * promise that resolves to a tuple containing the configuration object and
+ * AWS credentials (if applicable).
+ */
 const esConfig = async (host, metrics = false) => {
   let config;
+  let credentials;
   if (inTestMode() || 'LOCAL_ES_HOST' in process.env) {
     config = esTestConfig();
   } else if (metrics) {
     config = esMetricsConfig();
   } else {
-    config = await esProdConfig(host);
+    credentials = await getAwsCredentials();
+    config = esProdConfig(host, credentials);
   }
-  return config;
+  return [config, credentials];
 };
 
+/**
+ * `EsClient` is a class for managing an Elasticsearch client.
+ *
+ * @property {string} host - The host URL for the Elasticsearch instance.
+ * @property {boolean} metrics - A flag indicating whether metrics are enabled.
+ * @property {elasticsearch.Client} _client - The Elasticsearch client instance.
+ *
+ * @method constructor - Initializes a new instance of the `EsClient` class.
+ * @method initializeEsClient - Initializes the Elasticsearch client (this._client/client)
+ * if it hasn't been initialized yet.
+ * @method refreshClient - Refreshes the Elasticsearch client if the AWS credentials have changed,
+ * by creating a new Elasticsearch `Client` instance.
+ * @method client - Getter that returns the current Elasticsearch Client
+ */
```

Comment:
should esCLient also handle esConnectionsClient? why is is Search still used on line 255 of api/endpoints/collections.js?

Reply:
Because `Search` isn't really an EsClient class.     Part of the issue with the original implementation is we utilized a static class method to generate an Elasticsearch Client 

Search/BaseSearch are intended to provide cumulus-specific ES queries.   In this case, search is being used to allow:

`await esCollectionsClient.exists(collectionId))`

However, line 278 uses a method that's expecting an EsClient

`await deleteCollection({`

I'm assuming you're asking if we should pass in Search.client instead?      Or provide a way to set esClient *to* the new Search object's client param on 255? 

That's a reasonable question/maintenance refactor.   The *current* code as written is already using a separate Search.es() client, but there's no reason we need to EsClients in memory.    I'd be open to that as a separate PR, or the conversation if you feel it's a needed improvement while we're doing this work.    We just need to keep in mind that the removal of all of this is pending as well.

Reply:
ok, fair enough, I keep forgetting that this is all on the way out and maybe shouldn't be updated too hard. something along those lines is what I was getting at yea